src/main/scala/au/com/titanclass/streams/telemetry/MetricsJsonProtocol.scala

```diff
@@ -85,14 +85,14 @@ object MetricsJsonProtocol extends DefaultJsonProtocol {
     override def write(obj: MetricsReporter.MetricsSnapshot): JsValue =
       JsObject(
         "gauges" -> JsObject(
-          obj._1.asScala
+          obj._1.asScala.view
             .mapValues(g => new BigDecimalGauge(BigDecimal(g.getValue.toString)).toJson)
             .toMap
         ),
-        "counters"   -> JsObject(obj._2.asScala.mapValues(_.toJson).toMap),
-        "histograms" -> JsObject(obj._3.asScala.mapValues(_.toJson).toMap),
-        "meters"     -> JsObject(obj._4.asScala.mapValues(_.toJson).toMap),
-        "timers"     -> JsObject(obj._5.asScala.mapValues(_.toJson).toMap)
+        "counters"   -> JsObject(obj._2.asScala.view.mapValues(_.toJson).toMap),
+        "histograms" -> JsObject(obj._3.asScala.view.mapValues(_.toJson).toMap),
+        "meters"     -> JsObject(obj._4.asScala.view.mapValues(_.toJson).toMap),
+        "timers"     -> JsObject(obj._5.asScala.view.mapValues(_.toJson).toMap)
```

Comment:
I've never actually bothered to understand Scala's views until now! Scala 2.13 makes them a first-class citizen.

Reply:
Yes, I had the same thing the other day doing a groupBy on a Seq in 2.13 - deprecation notices all over the place until I swapped to use mapValues on the view